ocaml/runtime/compact.c

```diff
@@ -308,7 +308,7 @@ static void do_compaction (intnat new_allocation_policy)
         if (q != 0 && Is_white_hd (q)){
           size_t sz = Bhsize_hd (q);
           char *newadr = compact_allocate (sz);
-          memmove (newadr, p, sz);
+          if (newadr != (char*) p) memmove (newadr, p, sz);
```

Comment:
Is this for performance? The same check is generally present in memmove too, so not sure why this would help.

Reply:
It just seemed like a lot of pointless function calls, but I'm not particularly fussed about this change.